### PR TITLE
Created initial IotDev solution

### PR DESCRIPTION
**Pull request by shaggygi**
This is the beginning part of the [proposal to create a new IotDev CLI](https://github.com/dotnet/iot/issues/113).  This is nothing more than a new solution along with empty projects (core and tests) to get started.  Both projects have the [experimental package](https://github.com/dotnet/command-line-api/wiki/Your-first-app-with-System.CommandLine#install-the-systemcommandlineexperimental-package) added, as well.

The hope is to go ahead and check-in as a starter where future PRs will be created to add more functionality.

### CODE REVIEW

**Comment by joperezr** [comment]
What advantage does this one have over the CLI tool we already have? I ask because most of the surface area is already exposed through that tool, plus you recently added the ability to compile/execute scripts so I'm not sure I see the value on adding another tool that simply uses a different parser.

**Comment by shaggygi** [comment]
This is more of a longer term strategy proposal.  I think DeviceApiTester is very useful for now while System.Device.* APIs mature.  DeviceApiTester has helped me get up-to-speed quickly for 1) learning to deploy dev board apps, 2) troubleshoot the API and 3) create more helpful commands.

I don't think DeviceApiTester will go away anytime soon.  I have another [PR in for I2cDetect and I2cDump](https://github.com/dotnet/iot/pull/120)... and have more to come (only so many minutes in a day to code 😄).

However, I'm trying to think of a "bigger and longer" term vision where the proposal (IotDev) could grow with 1) better functionality and 2) support from a better command-line.  After all, @bitcrazed (Rich Turner) and team are apparently supporting this.  https://twitter.com/richturn_ms/status/1076183040848650240


I added a [list of a few areas](https://github.com/dotnet/iot/issues/113) to research while System.CommandLine matures.  Both DeviceApiTester and IotDev would live together for some time.  Eventually, we would deprecate DeviceApiTester.

IMHO, from my little time reviewing where System.CommandLine is headed, it seems like the better long-term approach to have a world-class CLI for interacting with System.Device.* APIs.

So long story short... this PR was nothing more than an empty solution to start a "playground" to learn about System.CommandLine and see what capabilities it will offer.

**Comment by joperezr** [comment]
> this PR was nothing more than an empty solution to start a "playground" to learn about System.CommandLine and see what capabilities it will offer.

I see, I think that this is probably not the best place for that. Perhaps a good direction for this PR, would be to move the existing tool to System.CommandLine?

**Comment by shaggygi** [comment]
> I see, I think that this is probably not the best place for that. Perhaps a good direction for this PR, would be to move the existing tool to System.CommandLine?

My [original issue](https://github.com/dotnet/iot/issues/113) created was to migrate DeviceApiTester to System.CommandLine.  However, after thinking about it more....

1. Its about branding.  DeviceApiTester doesn't sound and look as nice as IotDev (or **iotdev** in the console).  But that's just my opinion.
2. I think System.CommandLine needs more time grow before fully utilizing.  That's why I thought it would be easier to have both DeviceApiTester and IotDevice live in parallel until System.CommandLine was ready.  If we are able to have both for a temporary period, we just need to point out the intention so new users understand direction.  I don't think peeps would be too vested in DeviceApiTester and wouldn't mind the change once everything is determined.

If you still think it is a bad idea to house both at the same time, let's cancel this PR and keep the [original issue](https://github.com/dotnet/iot/issues/113) open.  I'll try to continue researching and adding content/notes.  We can circle back with another PR when ready.

**Comment by joperezr** [comment]
To be honest I don't think it would be good to have both tools for the same purpose where the only difference is how we parse arguments. If the problem is around branding (which I agree that perhaps DeviceApiTester isn't the best) we can rename, and I think that once System.CommandLine is more mature we can migrate the existing solution to use that instead.

**Comment by shaggygi** [comment]
Sounds reasonable.  I'll go ahead and kill this PR then.  In the meantime, I'll play around with System.CommandLine and provide some feedback on the original issue as time allows.
